### ISSUE TITLE
Trilinos as TPL

### ISSUE THREAD

**Issue by aprokop** [ORNL-CEES/DataTransferKit#468]
In order to make progress on #451, we should decide on the following questions:

1. Do we want to build DTK only outside of Trilinos, or do we want to allow building within?
2. [Depends on 1.] Do we want to use TriBITS or pure CMake?

**Comment by aprokop** [comment]
An additional question is to decide on the testing framework.

**Comment by sslattery** [comment]
I tend to agree with @Rombur here regarding building outside of trilinos and using cmake. I'm fine with Gtest or Boost for testing - I'll defer to @dalg24 and @Rombur on the testing framework

**Comment by masterleinad** [comment]
`DataTransferKit` is going to be archived.
